### PR TITLE
Fixes for `one_hot` and `reverse_one_hot`

### LINKED ISSUE #212: `one_hot` (and perhaps reverse) has been inadvertently broken
**Describe the bug**
Some weird change has broken the implementation of `GANDLF.utils.one_hot` (see below for details).

**To Reproduce**
1. Download this file: [segmentation.nii.gz](https://github.com/CBICA/GaNDLF/files/7221986/segmentation-11.nii.gz)
2. Run the following code:
```python
import SimpleITK as sitk
from GANDLF.utils import one_hot
import torch

seg_array = sitk.GetArrayFromImage(sitk.ReadImage("C:/path/to/segmentation.nii.gz"))
class_list=[0,"1||2"]
seg_tensor = torch.from_numpy(seg_array)
seg_tensor = seg_tensor.unsqueeze(0) # we do this to ensure batch dimensions is added
print(seg_tensor.shape)
seg_array_oh = one_hot(seg_tensor,class_list)
print(seg_array_oh.shape)
```
3. Which gives:
```powershell
torch.Size([1, 466, 512, 512])
torch.Size([1, 2, 512, 512])
```
4. See subsequent steps exploding.

**Expected behavior**
```python
print(seg_array_oh.shape)
```
Should give:
```powershell
torch.Size([1, 2, 466, 512, 512])
```

**Screenshots**
N.A.

**GaNDLF Version**
0.0.12-dev

**Desktop (please complete the following information):**
N.A.

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes #212

## Proposed Changes

- `one_hot` and related functionality fixed
- RGB check and fix for label has been added

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (would not break existing functionality): if changes breaks current code, please provide as many details as possible
- [x] Function/class source code documentation added/updated
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [x] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [x] If adding a submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [x] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [x] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation

### CODE REVIEW

**Review by Geeks-Sid — APPROVED** [review]

**Comment by sarthakpati** [comment]
Changes are working as expected in my tests. @iskobleva please open up a bug request if you see any more issues.
